### ISSUE TITLE
Update logging framework based on new version of PSFramework

### ISSUE THREAD

**Issue by andreasjordan** [dataplat/dbatools#8392]
### Summarize Functionality

See: https://twitter.com/FredWeinmann/status/1533198245496758274

> Yo fellows, the latest version of the [#psframework](https://twitter.com/hashtag/psframework?src=hashtag_click) [#powershell](https://twitter.com/hashtag/powershell?src=hashtag_click) module shipped today!
Massively improved logging performance for the logfile provider (and no more collision with your AV), new command to clear in-memory log and (thanks [@adamdriscoll](https://twitter.com/adamdriscoll) !) fixed a memory leak.

### Is there a command that is similiar or close to what you are looking for?

No

**Comment by adamdriscoll** [comment]
@FriedrichWeinmann probably has a good idea on this but from what I can tell, it looks like dbatools doesn't have the same problem that PSFramework had in terms of the memory leak. I can't speak to the logging performance.

**Comment by FriedrichWeinmann** [comment]
Indeed, for dbatools, this is not a memory leak issue, however the throughput improvements remain real

**Comment by andreasjordan** [comment]
Do you have time to work on this?

**Comment by wsmelton** [comment]
What needs to be fixed? We have no memory leak on our logging mechanism.

The logging system we have is the beginnings of what PSFramework became and at this point we are so far behind where that module has gone it is unrealistic to think Fred will have time for porting that over. It would be more efficient to swap our logging to depend on PSFramework...but that is not something I would agree with incorporating in the module.

In the area of performance if you want the module to perform better the best option is disabling our logging system itself. You should do that for any automation process, logging is for when you are debugging or troubleshooting.

**Comment by andreasjordan** [comment]
Thanks for the feedback, will close this now.

**Comment by potatoqualitee** [comment]
I think it's a fair question to ask if the fix is quick. Any perf improvements are welcomed

**Comment by FriedrichWeinmann** [comment]
I've added it for my vacation coding list (calendar blocker and everything so I don't forget).
Will do the upgrade - the overall effort should be doable.

**Comment by andreasjordan** [comment]
Hi @FriedrichWeinmann  - is this still on your to do list?

**Comment by andreasjordan** [comment]
Let me ping @FriedrichWeinmann again...

**Comment by andreasjordan** [comment]
Let's close this as there is no progress...
